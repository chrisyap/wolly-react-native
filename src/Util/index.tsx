import { Dimensions, Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';

const isAndroid = Platform.OS === 'android';
const isIos = Platform.OS === 'ios';
const isSmallScreen = Dimensions.get('window')?.width <= 360;
const getTestIdProp = (id?: string): object | null =>
  id ? (isIos ? { testID: id } : { accessibilityLabel: id }) : null;
const isTablet = DeviceInfo.isTablet();

const convertToPercent = (value: number): string => {
  return `${value * 100}%`;
};

export { convertToPercent, getTestIdProp, isAndroid, isIos, isSmallScreen, isTablet };
