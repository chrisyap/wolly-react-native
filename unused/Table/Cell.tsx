import React from 'react';
import V, { VProps } from '../View';

interface CellProps extends VProps {
  align?: 'right' | 'center' | 'left';
}

const Cell: React.FC<CellProps> = props => (
  <V
    py={props.py || 2}
    px={props.px || 4}
    flexDirection="row"
    justifyContent={props.align === 'right' ? 'flex-end' : props.align === 'center' ? 'center' : 'flex-start'}
    {...props}
  />
);

export { Cell };
