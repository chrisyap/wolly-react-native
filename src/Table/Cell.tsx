import React from 'react';
import V, { VProps } from '../View';
import { isTablet } from '../Util';

interface CellProps extends VProps {
  align?: 'right' | 'center' | 'left';
}

const Cell: React.FC<CellProps> = (props) => (
  <V
    py={props.py || isTablet ? 2 : 1}
    px={props.px || 2}
    flexDirection="row"
    justifyContent={props.align === 'right' ? 'flex-end' : props.align === 'center' ? 'center' : 'flex-start'}
    {...props}
  />
);

export { Cell };
