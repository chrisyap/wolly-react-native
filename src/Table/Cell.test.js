import React from 'react';
import { Cell } from './Cell';
import * as util from '../Util';

jest.mock('../Util');

const isTablet = (utility = util, boolean) => {
  utility.isTablet = boolean;
};

describe('Cell', () => {
  beforeEach(() => {
    isTablet(util, false);
  });

  const shallowWrapper = (props) =>
    shallow(<Cell {...props} />)
      .dive()
      .dive()
      .dive();

  const defaultProps = {
    children: 'Cell text',
  };

  it('renders correctly', () => {
    const wrapper = shallowWrapper(defaultProps);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders right aligned correctly', () => {
    const props = {
      ...defaultProps,
      align: 'right',
    };
    const wrapper = shallowWrapper(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders center aligned correctly', () => {
    const props = {
      ...defaultProps,
      align: 'center',
    };
    const wrapper = shallowWrapper(props);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Tablet', () => {
    beforeEach(() => {
      isTablet(util, true);
    });

    it('renders correctly', () => {
      isTablet(util, true);
      const wrapperTablet = shallowWrapper(defaultProps);
      expect(wrapperTablet).toMatchSnapshot();
    });

    it('renders right aligned correctly', () => {
      const props = {
        ...defaultProps,
        align: 'right',
      };
      isTablet(util, true);
      const wrapperTablet = shallowWrapper(props);
      expect(wrapperTablet).toMatchSnapshot();
    });

    it('renders center aligned correctly', () => {
      const props = {
        ...defaultProps,
        align: 'center',
      };

      isTablet(util, true);
      const wrapperTablet = shallowWrapper(props);
      expect(wrapperTablet).toMatchSnapshot();
    });
  });
});
