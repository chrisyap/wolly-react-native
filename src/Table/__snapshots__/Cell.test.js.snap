// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell Tablet renders center aligned correctly 1`] = `
<View
  align="center"
  flexDirection="row"
  justifyContent="center"
  px={2}
  py={2}
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "stretch",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      },
    ]
  }
>
  Cell text
</View>
`;

exports[`Cell Tablet renders correctly 1`] = `
<View
  flexDirection="row"
  justifyContent="flex-start"
  px={2}
  py={2}
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "stretch",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      },
    ]
  }
>
  Cell text
</View>
`;

exports[`Cell Tablet renders right aligned correctly 1`] = `
<View
  align="right"
  flexDirection="row"
  justifyContent="flex-end"
  px={2}
  py={2}
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "stretch",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      },
    ]
  }
>
  Cell text
</View>
`;

exports[`Cell renders center aligned correctly 1`] = `
<View
  align="center"
  flexDirection="row"
  justifyContent="center"
  px={2}
  py={1}
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "stretch",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
      },
    ]
  }
>
  Cell text
</View>
`;

exports[`Cell renders correctly 1`] = `
<View
  flexDirection="row"
  justifyContent="flex-start"
  px={2}
  py={1}
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "stretch",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
      },
    ]
  }
>
  Cell text
</View>
`;

exports[`Cell renders right aligned correctly 1`] = `
<View
  align="right"
  flexDirection="row"
  justifyContent="flex-end"
  px={2}
  py={1}
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "stretch",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
      },
    ]
  }
>
  Cell text
</View>
`;
